src/Domain/Creators/TimeDependence/CubicScale.cpp

```diff
@@ -69,32 +72,42 @@ CubicScale<MeshDim>::block_maps(const size_t number_of_blocks) const {
 template <size_t MeshDim>
 std::unordered_map<std::string,
                    std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>
-CubicScale<MeshDim>::functions_of_time() const {
+CubicScale<MeshDim>::functions_of_time(
+    const std::unordered_map<std::string, double>& initial_expiration_times)
+    const {
   std::unordered_map<std::string,
                      std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>
       result{};
-  const double initial_expiration_time =
-      initial_expiration_delta_t_ ? initial_time_ + *initial_expiration_delta_t_
-                                  : std::numeric_limits<double>::max();
 
+  // Functions of time don't expire by default
+  std::unordered_map<std::string, double> expiration_times{};
+  for (auto& cubic_name : functions_of_time_names_) {
+    expiration_times[cubic_name] = std::numeric_limits<double>::infinity();
+  }
+
+  // If we have control systems, overwrite these expiration times with the ones
+  // supplied by the control system
+  for (auto& [name, expr_time] : initial_expiration_times) {
+    if (expiration_times.count(name) == 1) {
```

Comment:
Since you deleted this `if` in the domain creator, also carry it through here

Reply:
For the BinaryCompactObjectDomain, it made sense to get rid of this `if` because it is the final stop for all expiration times. However, in the time dependences it isn't.

Consider a composition between cubic scale and rotation where only rotation is controlled. `initial_expiration_times` would only contain "Rotation". However, if I remove this `if`, CubicScale would still try to look for its names in `initial_expiration_times` (and fail) because it expects them to be there. Thus, I think this `if` should remain in all the TimeDependences